### PR TITLE
Changed Optimism L1 Oracle to support Mantle

### PR DESCRIPTION
**Pull request by ma33r**
### What
- Changes our existing `op_l1_oracle` to support the Mantle Network. Specifically, for Mantle, the `tokenRatio` parameter is retrieved and used to calculate a gas price
- Closes [BCI-3964](https://smartcontract-it.atlassian.net/browse/BCI-3964?atlOrigin=eyJpIjoiNzFjMGQyZDI0ZWE1NDIxMGJmMzVlM2MxYjcyMmYzY2EiLCJwIjoiaiJ9)

## Why
- Mantle's gas price formula is similar to that of Optimism Bedrock, but it also includes a parameter called tokenRatio. The docs detailing this can be found [here](https://docs-v2.mantle.xyz/devs/concepts/tx-fee/overviews#l2-gasprice---depends-on-the-transaction-type)

[BCI-3964]: https://smartcontract-it.atlassian.net/browse/BCI-3964?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: core/chains/evm/config/chaintype/chaintype.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ const (
 	ChainXLayer          ChainType = "xlayer"
 	ChainZkEvm           ChainType = "zkevm"
 	ChainZkSync          ChainType = "zksync"
+	ChainMantle          ChainType = "mantle"
```
Comment (amit-momin) [inline core/chains/evm/config/chaintype/chaintype.go]: nit: Think this follow alphabetical order too

File: core/chains/evm/gas/rollups/op_l1_oracle.go [hunk 2]
```diff
@@ -187,6 +192,16 @@ func newOpStackL1GasOracle(lggr logger.Logger, ethClient l1OracleClient, chainTy
...
 	}
 
+	// Encode calldata for tokenRatio method
+	tokenRatioMethodAbi, err := abi.JSON(strings.NewReader(MantleTokenRatioAbiString))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse GasPriceOracle %s() method ABI for chain: %s; %w", decimalsMethod, chainType, err)
+	}
+	tokenRatioCalldata, err := tokenRatioMethodAbi.Pack(tokenRatioMethod)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse GasPriceOracle %s() calldata for chain: %s; %w", decimalsMethod, chainType, err)
```
Comment (amit-momin) [inline core/chains/evm/gas/rollups/op_l1_oracle.go]: Think the error message got copied over so we should update the method
```suggestion
		return nil, fmt.Errorf("failed to parse GasPriceOracle %s() method ABI for chain: %s; %w", tokenRatioMethod, chainType, err)
	}
	tokenRatioCalldata, err := tokenRatioMethodAbi.Pack(tokenRatioMethod)
	if err != nil {
		return nil, fmt.Errorf("failed to parse GasPriceOracle %s() calldata for chain: %s; %w", tokenRatioMethod, chainType, err)
```

File: core/chains/evm/gas/rollups/op_l1_oracle.go [hunk 3]
```diff
@@ -443,6 +463,33 @@ func (o *optimismL1Oracle) getV1GasPrice(ctx context.Context) (*big.Int, error)
...
+	}, nil)
+	if err != nil {
+		return nil, fmt.Errorf("tokenRatio() call failed: %w", err)
+	}
+
+	// create bigInt from returned bytes
+	tokenRatio := new(big.Int).SetBytes(b)
+
+	// call optimism bedrock gas price function
+	v1GasPrice, err := o.getV1GasPrice(ctx)
```
Comment (amit-momin) [inline core/chains/evm/gas/rollups/op_l1_oracle.go]: Just realized that the OP V1 gas price method is only really calling `l1BaseFee()`. Could we potentially just do a batch call here for both `tokenRatio()` and `l1BaseFee90` so we can get the results in just one RPC call? The `getEcotoneFjordGasPrice` method should have an example of how that would look.

File: core/chains/evm/config/chaintype/chaintype.go (hunk 1 above)
Comment (ma33r) [reply core/chains/evm/config/chaintype/chaintype.go]: Thanks, just addressed this in the latest commit

File: core/chains/evm/gas/rollups/op_l1_oracle.go (hunk 2 above)
Comment (ma33r) [reply core/chains/evm/gas/rollups/op_l1_oracle.go]: Thank you, I just addressed this in the latest commit

File: core/chains/evm/gas/rollups/op_l1_oracle.go (hunk 3 above)
Comment (ma33r) [reply core/chains/evm/gas/rollups/op_l1_oracle.go]: Sounds good! I just addressed this in the latest commit

File: core/chains/evm/gas/rollups/op_l1_oracle_test.go [hunk 4]
```diff
@@ -111,6 +111,54 @@ func TestOPL1Oracle_ReadV1GasPrice(t *testing.T) {
 	}
 }
 
+func TestOPL1Oracle_ReadMantleGasPrice(t *testing.T) {
+	l1BaseFee := big.NewInt(100)
+	tokenRatio := big.NewInt(40)
+	oracleAddress := common.HexToAddress("0x1234").String()
+
+	t.Parallel()
+	t.Run("correctly fetches gas price if chain is Mantle", func(t *testing.T) {
```
Comment (amit-momin) [inline core/chains/evm/gas/rollups/op_l1_oracle_test.go:120]: Could you also add some simple test cases for RPC errors? I think we have some for Ecotone and Fjord for reference.

Comment (ma33r) [reply core/chains/evm/gas/rollups/op_l1_oracle_test.go:120]: Thanks, I just addressed this in the latest commit

**Review by amit-momin — APPROVED** [review]
LGTM

**Comment by lukaszcl** [comment]
@ma33r please merge develop to this branch (click "Update branch"). Otherwise the PR may be stuck in merge queue. This is due to recent CI changes made on develop.

**Review by huangzhen1997 — APPROVED** [review]
lgtm

**Review by prashantkumar1982 — APPROVED** [review]

**Pull request merged**